apps/code-review/src/app/pages/overview/services/open-directory.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable, inject } from '@angular/core';
+import { Observable, filter, map } from 'rxjs';
+import { OpenDirectoryState } from '../states';
+
+@Injectable()
+export class OpenDirectoryService {
+    private readonly state = inject(OpenDirectoryState);
+
+    public setOpenDirectoryState(fileFullPath: string, state: boolean): any {
+        this.state.set({
+            ...this.state.data,
+            [fileFullPath]: state
+        });
+    }
+
+    public getOpenDirectoryState(fileFullPath: string): Observable<boolean> {
```

Comment:
```suggestion
    public get(fileFullPath: string): Observable<boolean> {
```

Можливо не прямо `get`, але варто спростити назву методу. Вже є контекст (назва класу) і не треба дублювати слова в назві методів